server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -242,19 +328,35 @@ private boolean isDataStreamIndex(String indexName) {
      * @return If present, an error message to be given as the reason for failing
      * an operation. If empty, a sign that the operation is valid.
      */
-    public Optional<String> checkShardLimit(int newShards, ClusterState state) {
-        return checkShardLimit(newShards, state, getShardLimitPerNode(), getShardLimitPerCluster());
+    public Optional<String> checkShardLimit(int newShards, ClusterState state, RoutingPool shardPool) {
+        return shardPool == RoutingPool.REMOTE_CAPABLE
+            ? checkShardLimit(
+                newShards,
+                state.getMetadata().getTotalOpenRemoteCapableIndexShards(),
+                getRemoteCapableShardLimitPerNode(),
+                getRemoteCapableShardLimitPerCluster(),
+                state.getNodes().getWarmNodes().size(),
+                "Warm"
```

Comment:
instead of string , can we use `RoutingPool` itself ?  while printing the error message , we can convert into string .